src/Compilers/Core/Portable/InternalUtilities/MultiDictionary.cs

```diff
@@ -258,6 +258,20 @@ public bool Add(K k, V v)
             return true;
         }
 
+        public bool AddRange(K k, IEnumerable<V> values)
+        {
+            var result = false;
```

Comment:
The returned `bool` value doesn't match what I would have intuitively expected (I think `false` should be returned if any of the `Add` calls fails, instead of if any of the `Add` calls succeeds), but it could be that my expectation was incorrect. Is this modeled after some existing API?

Reply:
these are the right semantics.  The idea of `bool Set.Add` is that you can tell if the collection *changed* or not.  This can be used for fixed-point algorithms.  i.e. keep doing something as long as you are changing some result.  

I'm keeping that here.  As long as you were able to successfully add something, then the map was 'changed' and bool is returned to indicate that it was changed.  Returning false would indicate that adding all the items produced no change.

Reply:
okay. Thanks